apps/passport-client/components/screens/EmbeddedScreens/EmbeddedScreen.tsx

```diff
@@ -0,0 +1,67 @@
+import { PCDGetRequest, PendingPCD } from "@pcd/passport-interface";
+import { PCD, SerializedPCD } from "@pcd/pcd-types";
+import { assertUnreachable } from "@pcd/util";
+import { ReactNode, useCallback } from "react";
+import { useEmbeddedScreenState } from "../../../src/appHooks";
+import { EmbeddedScreenType } from "../../../src/embedded";
+import { useSyncE2EEStorage } from "../../../src/useSyncE2EEStorage";
+import { GenericProveScreen } from "../ProveScreen/GenericProveScreen";
+import { EmbeddedAddSubscription } from "./EmbeddedAddSubscription";
+
+/**
+ * EmbeddedScreen is used to control the UI when embedded in an iframe.
+ */
+export function EmbeddedScreen(): ReactNode {
+  const embeddedScreen = useEmbeddedScreenState();
+  useSyncE2EEStorage();
+  if (!embeddedScreen) {
+    return null;
+  }
+  if (embeddedScreen.screen?.type === EmbeddedScreenType.EmbeddedGetRequest) {
+    return (
+      <EmbeddedGetRequest
+        request={embeddedScreen.screen.request}
+        callback={embeddedScreen.screen.callback}
+      />
+    );
+  } else if (
+    embeddedScreen.screen?.type === EmbeddedScreenType.EmbeddedAccessRequest
+  ) {
+    return <div>Embedded Access Request</div>;
```

Comment:
is this ever visible / displayed to a user?

Reply:
No, it's a placeholder but there's nothing that would cause it to be displayed yet.